cvat/apps/engine/views.py

```diff
@@ -1705,8 +1705,17 @@ class JobViewSet(viewsets.GenericViewSet, mixins.ListModelMixin, mixins.CreateMo
     mixins.RetrieveModelMixin, PartialUpdateModelMixin, mixins.DestroyModelMixin,
     UploadMixin, DatasetMixin, CsrfWorkaroundMixin
 ):
-    queryset = Job.objects.select_related('assignee', 'segment__task__data',
-        'segment__task__project', 'segment__task__annotation_guide', 'segment__task__project__annotation_guide',
+    queryset = Job.objects.select_related(
+        'assignee',
+        'segment__task__data',
+        'segment__task__project',
+        'segment__task__annotation_guide',
+        'segment__task__project__annotation_guide',
+        'segment__task__source_storage',
+        'segment__task__target_storage',
+    ).prefetch_related(
+        'segment__task__organization',
+        'segment__task__owner',
```

Comment:
May you explain why using prefetch_related is better here?

From my opinion it just makes extra requests (what in general requires more time) to the database and does not have benefits in comparison with select_related

IMHO we just need to define queryset "per view", leaving in defaults only those fields, necessary to IAM logic (e.g.  organization, owner)

Reply:
> May you explain why using prefetch_related is better here?
> From my opinion it just makes extra requests (what in general requires more time) to the database and does not have benefits in comparison with select_related

Yes, it does extra requests and does joining on the server side instead of doing it on the DB side, like in `select_related`. There are 2 "opposite" sides of prefetching - prefetch and join everything on the DB side and prefetch everything separately and join on the server side. Basically yes, we'd always prefer joining on the DB side, but separation can help in the following case: there is a small table, and its results are expected to be reused in many results. In this case it may be beneficial to fetch this small table separately and join it on the server side. Doing it on the DB side would significantly increase traffic by populating it with repeated entries. Probably, it's the case in my setup - I have just several users and orgs. I think the best strategy for optimizing this is actually measure the changes on the production server. Or, at least, take the statistics from it, to make the split decisions informed.

> IMHO we just need to define queryset "per view", leaving in defaults only those fields, necessary to IAM logic (e.g. organization, owner)

Probably, you meant "per endpoint usecase"? Ok, it makes sense, I agree.